amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/common/parser/CommonShapeParseContext.scala

```diff
@@ -0,0 +1,305 @@
+package amf.shapes.internal.spec.common.parser
+
+import amf.aml.internal.semantic.{
+  AnnotationSchemaValidator,
+  IgnoreAnnotationSchemaValidator,
+  SemanticExtensionsFacade,
+  SemanticExtensionsFacadeBuilder
+}
+import amf.core.client.scala.config.ParsingOptions
+import amf.core.client.scala.errorhandling.AMFErrorHandler
+import amf.core.client.scala.model.document.Fragment
+import amf.core.client.scala.model.domain.extensions.CustomDomainProperty
+import amf.core.client.scala.model.domain.{AmfObject, Shape}
+import amf.core.client.scala.parse.document.{EmptyFutureDeclarations, ParsedReference, ParserContext}
+import amf.core.internal.parser.Root
+import amf.core.internal.parser.domain.{Annotations, Declarations, SearchScope}
+import amf.core.internal.remote.{Oas30, Spec}
+import amf.core.internal.utils.{AliasCounter, QName}
+import amf.shapes.client.scala.model.domain.{AnyShape, CreativeWork, Example}
+import amf.shapes.internal.domain.metamodel.AnyShapeModel
+import amf.shapes.internal.spec.RamlWebApiContextType.{DEFAULT, RamlWebApiContextType}
+import amf.shapes.internal.spec.ShapeParserContext
+import amf.shapes.internal.spec.common.SchemaVersion
+import amf.shapes.internal.spec.contexts.JsonSchemaRefGuide
+import amf.shapes.internal.spec.jsonschema.parser
+import amf.shapes.internal.spec.jsonschema.parser.JsonSchemaSettings
+import amf.shapes.internal.spec.jsonschema.ref.{AstIndex, AstIndexBuilder, JsonSchemaInference, JsonSchemaParser}
+import amf.shapes.internal.spec.oas.parser.{Oas2Settings, Oas2ShapeSyntax, Oas3Settings, Oas3ShapeSyntax}
+import amf.shapes.internal.spec.raml.parser.{Raml08Settings, Raml08ShapeSyntax, Raml10Settings, Raml10ShapeSyntax}
+import amf.shapes.internal.validation.definitions.ShapeParserSideValidations.{
+  AnnotationSchemaMustBeAny,
+  MissingAnnotationSchema
+}
+import org.mulesoft.common.client.lexical.SourceLocation
+import org.yaml.model._
+
+import scala.collection.mutable
+
+class CommonShapeParseContext(
```

Comment:
Yes, it is an option to unify them but there was another implementation of the ShapeParserContext interface that I didn't want to mess. The context that implementes the interface is in an isolated module that shouldn't bring us troubles (semantic jsonld).

Reply:
WebApi works with shapes, my fear is to have an abstract method implemented twice.
For instance, the shapes declaration map (name->shape) should be exactly the same for shapes and web apis, right?
Other case could be documentation or examples, the auxiliary methods for those are the same for shapes and web api.
Basically, web api context should be shape context plus web api things, but all shape related logic shouldn't be different at web api level.

Reply:
The WebApiContext inherits the CommonShapeParserContext so it should never override its methods. In the current state, a web api context is a shape parser context because it inherits from the "common" shape context

Reply:
Changed it.